tests/server-snap/task.yaml

```diff
@@ -0,0 +1,26 @@
+summary: Check snap web servers
+environment:
+  SNAP_NAME/pythonServer: xkcd-webserver
+  IP_VERSION/pythonServer: 4
+  PORT/pythonServer: 80
+  TEXT/pythonServer: XKCD rocks!
+  SNAP_NAME/goServer: go-example-webserver
+  IP_VERSION/goServer: 6
+  PORT/goServer: 8081
+  TEXT/goServer: Hello World
+prepare: |
+  sudo snap install $SNAP_NAME
+  cat > request.txt <<EOF
+  GET / HTTP/1.0
+
+  EOF
+  while ! netstat -lnt | grep -Pq "tcp.*?:$PORT +.*?LISTEN\n*"; do sleep 0.5; done
```

Comment:
Sorry, excuse me for not being clearer :) There are global `warn-timeout` and `kill-timeout` settings which can be customized [1], I've just pushed changes to redefine the `kill-timeout` for this task to 1 minute given the actions to be done. IMO it would be great if we can use the facilities from the framework and prevent as much as possible adding code, maybe in this case we could add a simple timeout checker for this concrete command, let me know what do you think.

Thanks!!

[1] https://github.com/snapcore/spread#timeouts